src/box/lua/schema.lua

```diff
@@ -3593,30 +3593,84 @@ local function role_check_grant_revoke_of_sys_priv(priv)
     end
 end
 
-box.schema.role.grant = function(user_name, ...)
+local function box_schema_role_grant_do(user_name, ...)
     local uid = role_resolve(user_name)
     if uid == nil then
         box.error(box.error.NO_SUCH_ROLE, user_name)
     end
     role_check_grant_revoke_of_sys_priv(...)
     return grant(uid, user_name, ...)
 end
-box.schema.role.revoke = function(user_name, ...)
+
+local function box_schema_role_revoke_do(user_name, ...)
     local uid = role_resolve(user_name)
     if uid == nil then
         box.error(box.error.NO_SUCH_ROLE, user_name)
     end
     role_check_grant_revoke_of_sys_priv(...)
     return revoke(uid, user_name, ...)
 end
-box.schema.role.info = function(role_name)
+
+local function box_schema_role_info_do(role_name)
     local rid = role_resolve(role_name)
     if rid == nil then
         box.error(box.error.NO_SUCH_ROLE, role_name)
     end
     return info(rid)
 end
 
+-- Wrap a function into transaction if none is active.
+local function atomic_wrapper(func)
+    return function(...)
+        -- No reason to start a transaction of one is active already.
+        if box.is_in_txn() then
+            return func(...)
+        else
+            return box.atomic(func, ...)
+        end
+    end
+end
+
+box.schema.space.create = atomic_wrapper(box_schema_space_create_do)
+box.schema.space.format = atomic_wrapper(box_schema_space_format_do)
+box.schema.space.upgrade = box_schema_space_upgrade_do
+box.schema.space.drop = atomic_wrapper(box_schema_space_drop_do)
+box.schema.space.rename = atomic_wrapper(box_schema_space_rename_do)
```

Comment:
`rename` and `format` and probably some other functions don't need the atomic wrapper because they do exactly one statement. Why did you decide to wrap them?

Reply:
Just followed the order to "Make [all](https://github.com/tarantool/tarantool/issues/4348) box.schema functions transactional". It's not required for some of them.

Reply:
The issue description says:

> Use `box.begin()` and `box.commit()` in all DDL functions exported to Lua, like `box.schema.space.create()` or `box.schema.user.grant()`.

`exists` and `info` aren't DDL functions.

Reply:
Dropped atomicity of simple single statement and read-only functions.